Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -478,19 +512,27 @@ private void HandleShuttleMovement(float frameTime)
                         totalForce += impulse;
                     }
 
-                    totalForce = shuttleNorthAngle.RotateVec(totalForce);
-
                     var forceMul = frameTime * body.InvMass;
-                    var maxVelocity = (ShuttleComponent.MaxLinearVelocity - body.LinearVelocity.Length()) / forceMul;
 
-                    if (maxVelocity != 0f)
-                    {
-                        // Don't overshoot
-                        if (totalForce.Length() > maxVelocity)
-                            totalForce = totalForce.Normalized() * maxVelocity;
+                    var localVel = (-shuttleNorthAngle).RotateVec(body.LinearVelocity);
+                    var maxVelocity = ObtainMaxVel(localVel, shuttle); // max for current travel dir
+                    var maxWishVelocity = ObtainMaxVel(totalForce, shuttle);
+                    var properAccel = (maxWishVelocity - localVel) / forceMul;
 
-                        PhysicsSystem.ApplyForce(shuttleUid, totalForce, body: body);
-                    }
+                    var finalForce = Vector2.Dot(totalForce, properAccel.Normalized()) * properAccel.Normalized();
```

Comment:
I don't really understand the changes being made on lines 522-535 or why they're being made.

Reply:
In a previous writeup of this PR, I was looking at a concave expansion of the maximum velocity space, which is why the maximum velocity check is even there - instead of clamping speed immediately if the shuttle attempted to burn towards its velocity vector, it would attempt to pivot towards the wish velocity until it was back below the max. That part of the code is no longer present, but a different case is still relevant.
If the shuttle is trying to burn towards being stationary, it won't be impeded in any way, and if it decides to burn towards a different direction it will glide over to that direction's max velocity. This transformation makes the shuttle feel more realistic, but it's kinda fumbled by ObtainMaxVel math being wrong.